### PR TITLE
PUD-1496: Migrate to more appropriate statuses

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt [hunk 1]
```diff
@@ -92,70 +92,91 @@ class RecallTest {
...
+    val recall = recall.copy(
+      inCustodyAtBooking = true,
+      assessedByUserId = ::UserId.random().value,
+    )
+
+    assertThat(recall.status(), equalTo(Status.AWAITING_DOSSIER_CREATION))
+  }
+
+  @Test
+  fun `NOT In Custody Recall with assessedByUserId and assessedByUserId set returns status ASSESSED`() {
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt:115]: `with assessedByUserId and assessedByUserId set`?
`returns status ASSESSED`?  -> `returns status ASSESSED_NOT_IN_CUSTODY`.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt [hunk 2]
```diff
@@ -92,70 +92,91 @@ class RecallTest {
...
+      assessedByUserId = ::UserId.random().value,
     )
 
-    assertThat(recall.status(), equalTo(Status.RECALL_NOTIFICATION_ISSUED))
+    assertThat(recall.status(), equalTo(Status.AWAITING_DOSSIER_CREATION))
   }
 
   @Test
-  fun `NOT in Custody Recall with bookedByUserId, recallNotificationEmailSentDateTime & assignee set returns status RECALL_NOTIFICATION_ISSUED`() {
+  fun `NOT in Custody Recall with bookedByUserId, assessedByUserId & assignee set returns status ASSESSED_NOT_IN_CUSTODY`() {
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt:162]: Covered by previous, new, test with return of `ASSESSED_NOT_IN_CUSTODY ` hence delete as duplicate?

**Review by AndyNewton — DISMISSED** [review]
Functionally looks fine but bit of clean up possible in `RecallTest` status tests.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt:162]: subtly different as includes assignee?

Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/RecallTest.kt:162]: Tho assignee no longer considered for NIC, so deleted

**Review by AndyNewton — APPROVED** [review]

**Pull request merged**